ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -29,15 +30,20 @@ class PaywallActivityLauncher {
     /**
      * Creates a new PaywallActivityLauncher from an activity.
      */
-    constructor(activity: ComponentActivity, resultHandler: PaywallResultHandler) {
-        activityResultLauncher = activity.registerForActivityResult(PaywallContract(), resultHandler)
-    }
+    constructor(activity: ComponentActivity, resultHandler: PaywallResultHandler) :
+        this(activity as ActivityResultCaller, resultHandler)
 
     /**
      * Creates a new PaywallActivityLauncher from a fragment.
      */
-    constructor(fragment: Fragment, resultHandler: PaywallResultHandler) {
-        activityResultLauncher = fragment.registerForActivityResult(PaywallContract(), resultHandler)
+    constructor(fragment: Fragment, resultHandler: PaywallResultHandler) :
+        this(fragment as ActivityResultCaller, resultHandler)
+
+    /**
+     * Creates a new PaywallActivityLauncher from an [ActivityResultCaller] instance.
+     */
+    constructor(resultCaller: ActivityResultCaller, resultHandler: PaywallResultHandler) {
```

Comment:
Hmm considering both `ComponentActivity` and `Fragment` implement `ActivityResultCaller`, we could potentially remove them and just leave this constructor... Originally I thought to separate them so it's easier to understand, so I think it's better to leave them though... We can reconsider later.

Reply:
Good point, I removed the other 2 since they're redundant.

Reply:
@tonidero can you confirm this looks good to you?